src/IssueViz.Security.UnitTests/Taint/TaintList/TaintIssuesControlViewModelTests.cs

```diff
@@ -701,6 +724,44 @@ public void AnalysisInformation_IssuesChanged_RaisesPropertyChanged()
             testSubject.AnalysisInformation.Should().BeSameAs(analysisInformation);
         }
 
+        [TestMethod]
+        public void ActiveDocChanged_ExpectedServerTypeSetOnlyWhenItIsChanged()
+        {
+
+            var activeDocumentTrackerMock = new Mock<IActiveDocumentTracker>();
+            var sonarQubeServiceMock = new Mock<ISonarQubeService>();
+
+            var testSubject = CreateTestSubject(activeDocumentTracker: activeDocumentTrackerMock.Object, sonarQubeService:sonarQubeServiceMock.Object);
+            int eventCount = 0;
+            testSubject.PropertyChanged += (sender, args) => {
+                if (args is { PropertyName: nameof(testSubject.ServerType) })
+                {
+                    eventCount++;
+                }
+            };
+
+            // 1. SC
```

Comment:
1. always use different tests
2. I think it's better here to assert (null -> non null), (null -> null), (not null -> not null, same value), (not null -> not null, different value), and (not null -> null)